setup/install_gatling/20_foss_install/10_install_dev_repos.rb

```diff
@@ -0,0 +1,184 @@
+  repo_config_dir = 'tmp/repo_configs'
+
+  BASE_URL = 'http://builds.puppetlabs.lan'
+
+  def has_cent7_repo?(package, version)
+    cent7_uri = URI("#{BASE_URL}/#{package}/#{version}/repo_configs/rpm/pl-#{package}-#{version}-el-7-x86_64.repo")
+
+    response_code = Net::HTTP.start(cent7_uri.host, cent7_uri.port) do |http|
+      http.head(cent7_uri.path).code
+    end
+
+    if response_code != "200"
+      Beaker::Log.notify("Skipping #{package} version #{version} because it doesn't appear to have a cent7 repo")
+      false
+    else
+      Beaker::Log.notify("Found Cent7 repo for #{package} version #{version}")
+      true
+    end
+  end
+
+  def get_cent7_repo(response_lines, package)
+    response_lines.
+        map { |l| l.match(/^.*href="([^"]+)\/\?C=M&amp;O=D".*$/)[1] }.
+        find { |v| has_cent7_repo?(package, v) }
+  end
+
+  def get_latest_server_version(version)
+    response = Net::HTTP.get(URI(BASE_URL + '/puppetserver/?C=M&O=D'))
+
+    # Scrape the puppetserver repo page for available puppetserver builds and
+    # filter down to only ones matching the specified branch.  The list of builds
+    # is ordered from most recent to oldest.  The resulting list is passed along
+    # to the get_cent7_repo routine, which returns a URL for the first build which
+    # has a cent7 repo in it.
+    get_cent7_repo(
+        response.lines.
+            select { |l| l =~ /<td><a / }.
+            select { |l| l =~ /">.*#{version}.*\/<\/a>/}, "puppetserver")
+  end
+
+  def get_latest_agent_version
+    url = "http://builds.delivery.puppetlabs.net/passing-agent-SHAs/api/v1/json/report-master"
+
+    uri = URI.parse(url)
+    response = Net::HTTP.get_response(uri)
+
+    if response.code == "200"
+      json = JSON.parse(response.body)
+      @install_opts[:puppet_agent_commit] = json["suite-commit"].strip
+      @install_opts[:puppet_agent_version] = json["suite-version"].strip
+    else
+      Beaker::Log.notify("Unable to get last successful build from: #{url}, " +
+                             "error: #{response.code}, #{response.message}")
+      nil
+    end
+  end
+
+  def get_latest_release_agent_version
+    response = Net::HTTP.get(URI(BASE_URL + '/puppet-agent/?C=M&O=D'))
+
+    # Scrape the puppet-agent repo page for available puppet-agent builds and
+    # filter down to only released builds (e.g., 1.2.3) vs.
+    # some mergely/SHA version.  The list of builds is ordered from most recent to
+    # oldest.  The resulting list is passed along to the get_cent7_repo routine,
+    # which returns a URL for the first build which has a cent7 repo in it.
+    opts[:puppet_agent_version] = get_cent7_repo(
+        response.lines.
+            select { |l| l =~ /<td><a / }.
+            select { |l| l.match(/^.*href="(\d+\.\d+\.\d+)\/\?C=M&amp;O=D".*$/) },
+        "puppet-agent")
+  end
+
+  # Mostly copied from: install_puppet_agent_dev_repo_on in Beaker's foss_utils.rb
```

Comment:
is there a ticket to merge this up with beaker?

Reply:
No, instead we should be able to remove this and just use Beaker once we get the new AWS account with access to our network.